### ISSUE TITLE
Write vignette on dimension ordering

### ISSUE THREAD

**Issue by charlie-gallagher** [benrwoodard/adobeanalyticsr#140]
Dimension ordering can make a dramatic difference in the time it takes to run a query, and we've included this point in the documentation. 

However, I don't think we've mentioned how dimension ordering can change your results. We should outline in a vignette some guidelines for when dimension ordering matters, and when you can fiddle with it. At the very least, it deserves a comment in the documentation.

**Comment by gilliganondata** [comment]
I'm thinking what you're calling out is that the `top` value at each level is going to return the "top _x_" _at that level_. So, if you have `dim_a` and `dim_b` and are reporting `visits` and `top = c(5,5)`, then:

* **`dimensions = c("dim_a", "dim_b")`** would return (up to) 25 rows based on the top 5 values of `dim_a` _overall_, and then the top 5 values of `dim_b` _within each value for `dim_a`
* **`dimensions = c("dim_b", "dim_a")`** may feel like it would return the same thing, but just with the first two columns reversed...but it might not, because the `dim_b` values returned would be the top 5 values of `dim_b` _overall_, and then the top 5 values of `dim_a` _within each value for `dim_b`

(Um...okay...I see what you mean. This warrants some thought on how best to explain in the documentation _and_ warrants a vignette to illustrate! And...is this what you're  referencing?)

Do you have a couple of standard dimensions in mind that would be illustrative? I'm thinking (relatively) higher cardinality dimensions—`page` and `mobiledevicename`?

**Comment by charlie-gallagher** [comment]
Exactly, and that is a good start towards an explanation. It does deserve some thought to have a clear and simple example of when switching the order isn't the same. I don't have any standard dimensions in mind yet, but I'll think about it for a bit.
